tools/Google.Cloud.Tools.VersionCompat/Cause.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google.Cloud.Tools.VersionCompat
+{
+    public enum Cause
```

Comment:
I think we may end up wanting more fine-grained causes, or possibly hierarchical, if we're going to allow configuration to drive this. Obviously that's a longer-term issue though.